### PR TITLE
added design for sos tracker-1

### PR DESCRIPTION
**Pull request by yellowHatpro**
-basic design added for sos feature
-viewutil class implemented, added a basic toast func which can come in handy
few vector assets added

### CODE REVIEW

**Comment by yellowHatpro** [comment]
.idea file removed as well

File: app/src/main/res/layout/fragment_alert.xml [hunk 1]
```diff
@@ -1,12 +1,57 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/frameLayout"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
```
Comment (SunitRoy2703) [inline app/src/main/res/layout/fragment_alert.xml:7]: Add some padding here, maybe 10-15 dp

File: app/src/main/res/layout/fragment_alert.xml [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
+        app:layout_constraintTop_toBottomOf="@id/hellouser"
+        android:background="@drawable/ic_sos_contacts"
+        app:layout_constraintBottom_toTopOf="@id/sosbutton"/>
+
+    <ImageButton
+        android:id="@+id/ic_message"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/_50sdp"
+        app:layout_constraintTop_toBottomOf="@id/hellouser"
```
Comment (SunitRoy2703) [inline app/src/main/res/layout/fragment_alert.xml:39]: Keep position-related attributes at the end of the View tag

**Review by SunitRoy2703 — CHANGES_REQUESTED** [review]

File: app/src/main/res/drawable/ic_edit_sos_message.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (yellowHatpro) [reply app/src/main/res/drawable/ic_edit_sos_message.xml:1]: -redesigned edit message icon
-added circle outline and made icon smaller

File: app/src/main/res/drawable/ic_sos_contacts.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (yellowHatpro) [reply app/src/main/res/drawable/ic_sos_contacts.xml:1]: -added circle outline

File: app/src/main/res/layout/fragment_alert.xml (hunk 1 above)
Comment (yellowHatpro) [reply app/src/main/res/layout/fragment_alert.xml:7]: -added padding 15dp

File: app/src/main/res/layout/fragment_alert.xml (hunk 2 above)
Comment (yellowHatpro) [reply app/src/main/res/layout/fragment_alert.xml:41]: -sir can you check now is it fine now??
I still had some confusion around it a bit

File: app/src/main/res/layout/fragment_alert.xml [hunk 5]
```diff
@@ -2,56 +2,59 @@
...
-        android:layout_marginTop="@dimen/_100sdp"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:background="@drawable/ic_sos_button"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="@id/ic_message" />
+        app:layout_constraintTop_toTopOf="@id/ic_message"
+        android:layout_marginTop="@dimen/_100sdp"
```
Comment (SunitRoy2703) [inline app/src/main/res/layout/fragment_alert.xml:56]: Add this above ` app:layout_constraintBottom_toBottomOf="parent" `

**Review by SunitRoy2703 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_alert.xml (hunk 2 above)
Comment (SunitRoy2703) [reply app/src/main/res/layout/fragment_alert.xml:41]: It looks good now, let me know if you still have the doubt

File: app/src/main/res/drawable/ic_edit_sos_message.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SunitRoy2703) [reply app/src/main/res/drawable/ic_edit_sos_message.xml:1]: Great work!
The icon is looking better now, we can change it later if requires

File: app/src/main/res/layout/fragment_alert.xml (hunk 5 above)
Comment (yellowHatpro) [reply app/src/main/res/layout/fragment_alert.xml:56]: Sir app:layout_constraintBottom_toBottomOf="parent" is there on line 52 .. Do I have to change its position???

Comment (SunitRoy2703) [reply app/src/main/res/layout/fragment_alert.xml:56]: `android:layout_marginTop="@dimen/_100sdp"` this above `app:layout_constraintBottom_toBottomOf="parent"`

**Comment by yellowHatpro** [comment]
Done the required change sir

**Comment by SunitRoy2703** [comment]
> Done the required change sir

Great, I'm merging it!

**Pull request merged**